### ISSUE TITLE
a few more IRCClient docstrings

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3365]
|<img alt="benjamin's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| benjamin reported|
|-|-|
|Trac ID|trac#3365|
|Type|enhancement|
|Created|2008-07-23 23:35:42Z|

The attached patch adds a few more docstrings to IRCClient. Most of them are one-liners, but the generated docs did look awfully lonesome....

Attachments:

* [irc_client_docstrings.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/dc8/dc8bdad7c112bf00ff7f6c6dcbbf5b546546d270/64c31de5b50f74337032d2336124442ee6b62455.patch) (4864 bytes) - added by benjamin on 2008-07-23 23:36:02Z - 
* [irc_client_docstrings2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/dc8/dc8bdad7c112bf00ff7f6c6dcbbf5b546546d270/46b4761ec9d95da94a731d8c726dbb73d4af6c6d.patch) (11153 bytes) - added by benjamin on 2008-07-24 13:37:30Z - 
* [irc-fixes.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/dc8/dc8bdad7c112bf00ff7f6c6dcbbf5b546546d270/4be59bca5b7a5f9a09a20c5b2f378caa4122ed8a.patch) (12233 bytes) - added by exarkun on 2008-07-24 14:32:42Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__3365 3365
type__enhancement enhancement
reporter__benjamin benjamin
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__words words
keywords__ 
time__1216856142000000 1216856142000000
changetime__1217508877000000 1217508877000000
version__None None
owner__ 
cc__exarkun cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun set owner to benjamin|
|-|-|

Thanks!  This looks pretty good.  Could you do a couple more things?  The docstrings should be formatted like this:

```python
"""
Foo
"""
```

(I know, the rest of irc.py doesn't look like that, it's older than most of the coding standard).

And could you add epytext-formatted documentation for the parameters of these methods?

Stuff like:

```python
    def join(self, channel, key=None):
        """
        Join a channel.

        @type channel: C{str}
        @param channel: The name of the channel to join.  If it has no channel
            prefix, C{'#'} will be added.

        @type key: C{str} or L{NoneType}
        @param key: If specified, the key to use to join the channel.
        """
```

would be really great.

**Comment by twisted-trac** [comment]
|<img alt="benjamin's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|benjamin commented|
|-|-|

I hope this new patch does the trick...

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

Thanks, those changes are great.  I made a few more simple coding-standard updates.  Attaching a patch that includes all the changes.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra set owner to @exarkun|
|-|-|

Looks good, except for one instance of ```msg()``` in irc.py on line 927 that wasn't quoted with C{}

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set status to `closed`|
|-|-|

(In [24472]) Apply irc-fixes.patch, improving `IRCClient` docstrings

Author: benjamin, exarkun
Reviewer: thijs
Fixes: [#3365](https://github.com/twisted/twisted/issues/3365)

Reformat existing docstrings for `IRCClient` to meet the coding standard,
expand some of them to include more detailed information, and fix general
module-level coding standard violations.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
